cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -1163,6 +1175,118 @@ protected Future<Void> clusterId(KafkaStatus kafkaStatus) {
+                                                LOGGER.debugCr(reconciliation, "Got {} topic descriptions", topicDescriptions.size());
+                                                return Future.succeededFuture(brokerHasAnyReplicas(topicDescriptions, podId, kafkaStatus));
+                                            }).recover(error -> {
+                                                LOGGER.warnCr(reconciliation, "failed to get topic descriptions", error);
+                                                return Future.failedFuture(error);
+                                            });
+
+                                } catch (KafkaException e) {
+                                    LOGGER.warnCr(reconciliation, "Kafka exception getting clusterId {}", e.getMessage());
+                                } finally {
+                                    if (kafkaAdmin != null) {
+                                    }
+                                }
+                                future.complete();
+                            }, true, resultPromise);
+                    return resultPromise.future();
+                });
+    }
+
+
+    protected Future<Set<String>> topicNames(Admin kafkaAdmin) {
+        Promise<Set<String>> namesPromise = Promise.promise();
+        kafkaAdmin.listTopics(new ListTopicsOptions().listInternal(true)).names()
+                .whenComplete((names, error) -> {
+                    if (error != null) {
+                        namesPromise.fail(error);
+                    } else {
+                        LOGGER.debugCr(reconciliation, "Got {} topic names", names.size());
+                        namesPromise.complete(names);
+                    }
+                });
+        return namesPromise.future();
+    }
+
+    protected Future<Collection<TopicDescription>> describeTopics(Admin kafkaAdmin, Set<String> names) {
+        Promise<Collection<TopicDescription>> descPromise = Promise.promise();
+        kafkaAdmin.describeTopics(names).allTopicNames()
+                .whenComplete((tds, error) -> {
+                    if (error != null) {
+                        descPromise.fail(error);
+                    } else {
+                        LOGGER.debugCr(reconciliation, "Got topic descriptions for {} topics", tds.size());
+                        descPromise.complete(tds.values());
+                    }
+                });
+        return descPromise.future();
+    }
+
+    private Boolean brokerHasAnyReplicas(Collection<TopicDescription> tds, int podId, KafkaStatus kafkaStatus) {
+
+        for (TopicDescription td : tds) {
+            LOGGER.traceCr(reconciliation, td);
+            for (TopicPartitionInfo pd : td.partitions()) {
+                for (org.apache.kafka.common.Node broker : pd.replicas()) {
+                    if (podId == broker.id()) {
+                        canScaleDown = false;
+                        replicasBeforeScaleDown = currentReplicas;
```

Comment:
can you break the for loop when you already found that this broker cannot scale down because it hosts at least a partition replica? I guess there is no need to continue checking all other partitions, or?